rule/method/src/main/java/io/opentracing/contrib/specialagent/rule/method/MethodAgentRule.java

```diff
@@ -0,0 +1,74 @@
+package io.opentracing.contrib.specialagent.rule.method;
+
+import io.opentracing.contrib.specialagent.AgentRule;
+import net.bytebuddy.agent.builder.AgentBuilder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.dynamic.DynamicType;
+import net.bytebuddy.utility.JavaModule;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+/**
+ * MethodAgentRule
+ *
+ * @author code98@163.com
+ * @date 2019/11/15 5:27 下午
+ */
+public class MethodAgentRule extends AgentRule {
+
+    private static final String METHOD_CONFIG = "sa.config.instrumentation.plugin.method";
+
+    @Override
+    public Iterable<? extends AgentBuilder> buildAgent(AgentBuilder builder) throws Exception {
+        List<AgentBuilder> builders = new ArrayList<>();
+
+        String configInfos = System.getProperty(METHOD_CONFIG);
+        if (configInfos == null || "".equals(configInfos)) {
+            // sa.config.instrumentation.plugin.method is not config
+            return builders;
+        }
+
+        String[] classMethods = configInfos.split(";");
+        for (String classMethod : classMethods) {
+            String[] classMethodSplit = classMethod.split("#");
+            String className = classMethodSplit[0];
+            String method = classMethodSplit[1];
+
+            builders.add(
+                    builder.type(named(className)).transform(new AgentBuilder.Transformer() {
+                        @Override
+                        public DynamicType.Builder<?> transform(DynamicType.Builder<?> builder,
+                                                                TypeDescription typeDescription,
+                                                                ClassLoader classLoader,
+                                                                JavaModule module) {
+                            return builder.visit(Advice.to(MethodIntercept.class).on(named(method)));
```

Comment:
@code98, I am concerned that this visitor rule is "too open". Your `configInfos` spec does not allow one to provide a return type or argument types to be matched by the visitor. This will result in _all_  overloaded methods by the same name to be matched, which can lead to performance degradation.

Reply:
Hmm, you're right, this rule is "too open". There is no time this week to develop more advanced rules. Submit the first version of the code, and I will squeeze some time to develop the second version. So, is this OK?